src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala

```diff
@@ -93,8 +93,10 @@ object StreamingAOIMonitor
 
             val notifications: List[Notification] = queryNotifications(interval)
             notifications.foreach { println(_) }
-            val notificationsDf = notifications
-              .toDF("notificationId", "userId", "geom", "name", "email")
+            val notificationsRdd = spark.sparkContext.parallelize(notifications).map {
```

Comment:
Don't pack into an RDD.  Use `spark.createDataset(notifications)` (requires that `notifications` are filled with a case class instance).  You can then `groupByKey(_.notificationId)` or `join` using a more DataFrame-like interface.

Reply:
Updated. This is definitely cleaner and clearer.